### PR TITLE
Build macOS with compatibility independent of build environment macOS version

### PR DESCRIPTION
**Pull request by anthonybilinski**
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/6588)

### CODE REVIEW

File: .ci-scripts/build-osx-deps.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+#    Copyright © 2021 by The qTox Project Contributors
```
Comment (robinlinden) [inline .ci-scripts/build-osx-deps.sh]: >2021

File: .github/workflows/build-test-deploy.yaml [hunk 2]
```diff
@@ -551,6 +551,17 @@ jobs:
           replacesArtifacts: true
           token: ${{ secrets.GITHUB_TOKEN }}
           artifacts: "qTox-nightly.dmg,qTox-nightly.dmg.sha256"
+  build-osx-users:
+    name: macOS user
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (robinlinden) [inline .github/workflows/build-test-deploy.yaml:558]: There is a v3 of this now.

**Review by robinlinden — APPROVED** [review]

File: .ci-scripts/build-osx-deps.sh (hunk 1 above)
Comment (anthonybilinski) [reply .ci-scripts/build-osx-deps.sh]: Fixed.

File: .github/workflows/build-test-deploy.yaml (hunk 2 above)
Comment (anthonybilinski) [reply .github/workflows/build-test-deploy.yaml:558]: Ah good to know, will update all of them in a new PR.

Comment (anthonybilinski) [reply .github/workflows/build-test-deploy.yaml:558]: Update PR is here: https://github.com/qTox/qTox/pull/6600

**Pull request merged**